src/extensions/extensionsHelper.ts

```diff
@@ -138,21 +144,21 @@ export function substituteParams(original: object[], params: { [key: string]: st
  * @param paramSpec information on params parsed from extension.yaml
  * @return JSON object of params
  */
-export function populateDefaultParams(paramVars: any, paramSpec: any): any {
+export function populateDefaultParams(paramVars: Record<string, string>, paramSpecs: Param[]): any {
   const newParams = paramVars;
 
-  _.forEach(paramSpec, (env) => {
-    if (!paramVars[env.param]) {
-      if (env.default) {
-        newParams[env.param] = env.default;
-      } else if (env.required) {
+  for (const param of paramSpecs) {
```

Comment:
💯

Reply:
This is awesome - learned some new syntax today :)